components/PralineForm/index.js

```diff
@@ -0,0 +1,216 @@
+import useSWR from "swr";
+import { CldUploadButton } from "next-cloudinary";
+import Image from "next/image";
+import IngredientList from "./IngredientList";
+import InputField from "./InputField";
+
+export default function ProductForm() {
+  const [zutaten, setZutaten] = useState([]);
+  const [allergene, setAllergene] = useState([]);
+  const [currentPraline, setCurrentPraline] = useState(null);
+  const [nameField, setNameField] = useState("");
+  const [versionField, setVersionField] = useState("");
+  const [weightField, setWeightField] = useState("");
+  const [imageId, setImageId] = useState(null);
+
+  function handleAddIngredient(newIngredient) {
+    setZutaten([...zutaten, { ...newIngredient, id: uid() }]);
+  }
+
+  function handleDeleteIngredient(id) {
+    setZutaten(zutaten.filter((zutat) => zutat.id !== id));
+  }
+
+  function handleAddAllergen(allergeneMitMenge) {
+    setAllergene([...allergene, { ...allergeneMitMenge, id: uid() }]);
+  }
+
+  function handleDeleteAllergen(id) {
+    setAllergene(allergene.filter((allergen) => allergen.id !== id));
+  }
+
+  function cancel() {
+    setZutaten([]);
+    setAllergene([]);
+    setCurrentPraline(null);
+    setNameField("");
+    setVersionField("");
+    setWeightField("");
+  }
+
+  const { data, isLoading, mutate } = useSWR("/api/pralinen");
+  // console.log("data", data);
+
+  if (isLoading) {
+    return <h1>Loading...</h1>;
+  }
+
+  if (!data) {
+    return null;
+  }
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+    console.log("event", event);
+
+    if (zutaten.length === 0) {
+      return;
+    }
+
+    const formData = new FormData(event.target);
+    const pralineData = Object.fromEntries(formData);
+
+    const newPraline = {
+      name: pralineData.name,
+      version: pralineData.version,
+      weight: pralineData.weight,
+      imageId: imageId,
+      ingredients: zutaten.map((zutat) => ({
+        ingredient: zutat.ingredient,
+        amount: zutat.amount,
+      })),
+      allergyTraces: allergene.map((allergen) => ({
+        ingredient: allergen.ingredient,
+        amount: allergen.amount,
+      })),
+    };
+
+    console.log("newPraline", newPraline);
+
+    console.log("pralineData", pralineData);
+
+    const response = await fetch("/api/pralinen", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(newPraline),
+    });
+
+    if (response.ok) {
+      mutate();
+    }
+
+    setZutaten([]);
+    setAllergene([]);
+    setNameField("");
+    setVersionField("");
+    setWeightField("");
+    event.target.reset();
+    event.target.elements[0].focus();
+  }
+
+  async function handleDelete() {
+    await fetch(`/api/pralinen/${currentPraline._id}`, {
+      method: "DELETE",
+    });
+    setZutaten([]);
+    setAllergene([]);
+    setNameField("");
+    setVersionField("");
+    setWeightField("");
+    setCurrentPraline(null);
+  }
```

Comment:
`mutate` is missing here. After deleting an entry from the database, you need to update the app to reflect the change.